docs/threat_models/smart_contracts/multipool_autoswap/mpas_component.puml

```diff
@@ -0,0 +1,83 @@
+@startuml
+'https://plantuml.com/component-diagram
+header Agoric Operating Systems
+footer v0.0.1
+title Multiswap Auto Pool Component Diagram
+
+skinparam componentStyle rectangle
+skinparam defaultTextAlignment center
+
+actor "Multipool Autoswap Contract creator" as cc1
+actor "Random Pool Creator" as cc2
+actor "Person wanting to swap" as cc3
+interface "startInstance()" as si0
+
+package "Multipool Autoswap" {
+    [creatorFacet] as cf0
+    node contractFacet <<zcf>> as conf1 {
+        package "poolSeatA" as pa1 {
+           interface "incrementBy" as pa1i1
+           interface "decrementBy" as pa1i2
+        }
+
+        package "poolSeatB" as pa2 {
+            interface "incrementBy" as pa2i1
+            interface "decrementBy" as pa2i2
+        }
+        package "UserSeat" as use {
+            interface "incrementBy" as usi1
+            interface "decrementBy" as usi2
+        }
+
+        interface "reallocate" as reall
+
+    }
+    package "publicFacet" as pf0 {
+        interface "addPool()" as ap
+        interface "otherMethods..." as om
+        interface "offer()" as offer
```

Comment:
I think `offer` should be part of Zoe instead. Is there a reason to put it on the `publicFacet`?

Reply:
Technically, yes -- but in order to simplify the diagram and not bring the entirety of Zoe into this context, I opted to represent calling Zoe through the `publicFacet`. We should (and will) diagram all Zoe APIs in a separate diagram pending ongoing work, but this isn't the right place to do that just yet.

Reply:
Ok, it makes sense to me to leave out Zoe as an object in the diagram for simplification's sake. But, the diagram is going to be misleading in important ways if `offer` is on the public facet of the contract.  The viewer would think that contract code handles offer payments, which could violate offer safety and payout liveness, thus undermining Zoe entirely.  

So, this is something we need to change before this gets merged in. But, I think there's an easy fix, which is an asterisk + comment on the diagram saying that `offer` is actually part of Zoe, and no contract code has access to the user's payments.